File: frontend/src/events.js
// Canonical event names shared between Go and JS.
// Keep in sync with internal/events/events.go.

export const EVENTS = {
  // Clipboard lifecycle
  CLIPBOARD_UPDATED: 'clipboard-updated',

  // Sync
  SYNC_STATUS: 'sync-status',

  // Window visibility
  WINDOW_SHOWN:  'window-shown',
  WINDOW_HIDING: 'window-hiding',

// Navigation
  NAVIGATE: 'navigate',

  // Stack mode
  STACK_MODE_CHANGED: 'stack-mode-changed',
}

File: internal/events/events.go
// Package events defines the canonical event names shared between Go and JS.
//
// These constants replace raw string literals in both codebases. Changing a name
// here will break at compile time in Go and at import time in JS — no silent failures.
package events

// Clipboard lifecycle
const (
	ClipboardUpdated = "clipboard-updated"
)

// Sync
const (
	SyncStatus = "sync-status"
)

// Window visibility
const (
	WindowShown  = "window-shown"
	WindowHiding = "window-hiding"
)

// Navigation
const (
	Navigate = "navigate"
)

// Paste order (stack / queue / normal)
const (
	PasteOrderChanged = "paste-order-changed"
)

// Frontend log relay
const (
	FrontendLog = "frontend-log"
)
